R-package/tests/testthat/test_lgb.Booster.R

```diff
@@ -1653,6 +1654,21 @@ test_that("Booster's print, show, and summary work correctly", {
     .check_methods_work(model)
 })
 
+test_that("Booster's print, show, and summary work correctly when objective is not provided", {
+  data("iris")
+  model <- lgb.train(
+      data = lgb.Dataset(
+          as.matrix(iris[, seq_len(3L)])
+          , label = iris[, 4L]
+      )
+      , verbose = .LGB_VERBOSITY
+      , nrounds = 5L
+      , params = list(num_threads = .LGB_MAX_THREADS)
+  )
+
+  .check_methods_work(model)
```

Comment:
```suggestion
  .check_methods_work(model)
  log_txt <- capture.output({
     ret <- print(model)
   })
   expect_true(any(printed_txt == "Objective: (default)"))
```

`.check_methods_work()` is great and tests a lot of things, but for this particular PR I think it'd be helpful to **also** test that `print.lgb.Booster()` prints **exactly** what we expect for the objective.

Could you add something like this here?

Reply:
Sure thing!